src/main/java/org/veupathdb/service/eda/ss/model/reducer/StreamIntersectMerger.java

```diff
@@ -159,22 +164,31 @@ public PeekableIterator(Iterator<Long> stream) {
 
     /**
      * Advance the stream until the next element either matches or exceeds the key.
+     * If there are duplicates in the stream, advances the stream to the last duplicate.
      * @param key Value to advance the stream to or past if the key is not present in the stream.
      * @return Value the stream is advanced to, either key or the next highest value of the stream.
      */
     public Long skipUntilMatchesOrExceeds(Long key) {
       if (next == null) {
         return null;
       }
-      while (next < key) {
+      Long prev = null;
```

Comment:
I originally attempted what you suggested a while back and added this de-duping to the FilteredValueIterator. However, this required that class to operate in two different modes depending on whether we want to include duplicates or exclude duplicates (i.e. whether we're filtering or outputting values). It felt to me like this was a more natural place to put the logic since this class is responsible for reconciling and merging multiple streams of data. It might still warrant a separate implementation of the Merger for single vs. multi value for efficiency's sake and to keep it simpler in the case of single-value variables.